vulnfeeds/cves/versions.go

```diff
@@ -52,35 +53,115 @@ type CPE struct {
 	Other      string
 }
 
-func extractGitHubCommit(link string) *FixCommit {
-	// Example: https://github.com/google/osv/commit/cd4e934d0527e5010e373e7fed54ef5daefba2f5
-	u, err := url.Parse(link)
+// Returns the base repository URL for supported repository hosts.
+func repo(u string) (string, bool) {
+	parsedURL, err := url.Parse(u)
```

Comment:
Would it make more sense to have a single function for parsing a repo url, and return a struct containing both the repo and commit in separate fields?

Reply:
I debated this, but felt that for smaller units of testability to pursue the approach I took here.

Reply:
You could have both: 

you could have a underlying function that does both, which could simplify the implementation. Then have 2 convenience/public functions that only do 1.

Reply:
I think we need to align on requirements and callers for these functions. 

Right now, you can feed a URL, any URL to the `Repo()` function for the purposes of getting a known Git repo from it. That URL may or may not also align with being a Git commit (think GitHub Issue, GitHub Pull Request before we're able to resolve them to commits). There's a larger set of URLs than just Git commits, basically. There are going to be URLs where `Repo()` works and `Commit()` doesn't.

Should potential callers of `Repo()` be penalised because a Git commit wasn't derivable, but a base Git repository could?

Right now it's `extractGitCommit()`'s job to return the struct and originally `repo()` and `commit()` were individually testable implementation details for how that struct got constructed.

Reply:
The returned value of this combined function can simplify set `nil` on the fields it's not able to derive, which aligns with accounting for cases where `Repo()` works but `Commit()` doesn't. 

That said, there may be benefits in terms of error handling if we separate too. I don't feel strongly enough about this, so feel free to go either way for this.

Reply:
> The returned value of this combined function can simplify set `nil` on the fields it's not able to derive, which aligns with accounting for cases where `Repo()` works but `Commit()` doesn't.

Right now, you know if `Commit()` or `Repo()` works because you get your boolean back. Alternatively, the caller has to start digging around in structs to know if it got what it wanted, which seems more complicated to me?